src/Sign.Core/DataFormatSigners/ClickOnceSigner.cs

```diff
@@ -273,5 +274,53 @@ public void CopySigningDependencies(FileInfo deploymentManifestFile, DirectoryIn
                 }
             }
         }
+
+        /// <summary>
+        /// Try and find the application manifest (.manifest) file from a clickonce application manifest (.application / .vsto
+        /// There might not be one, if the user is attempting to only re-sign the deployment manifest without touching other files.
+        /// This is necessary because there might be multiple *.manifest files present, e.g. if a DLL that's part of the clickonce
+        /// package ships its own assembly manifest which isn't a clickonce application manifest.
+        /// </summary>
+        /// <param name="deploymentManifest"></param>
+        /// <returns>A string containing the file name of the Application manifest, or null if it couldn't be found.</returns>
+        /// <exception cref="InvalidDataException"></exception>
+        private string? GetApplicationManifestForDeploymentManifest(FileInfo deploymentManifest)
+        {
+            var xml = new XmlDocument();
+            xml.Load(deploymentManifest.FullName);
+            // there should only be a single result here, if the file is a valid clickonce manifest.
+            var dependentAssemblies = xml.GetElementsByTagName("dependentAssembly");
```

Comment:
We prefer using the explicit type if the implied type is not super obvious.  It will facilitate understanding both in a browser and IDE.